### ISSUE TITLE
Deadlock in `ParallelDataCollector`

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/libSplash#211]
It looks like there is a race condition in `DCDataSet::create()` if the [setChunking](https://github.com/ComputationalRadiationPhysics/libSplash/blob/c536da491a47522502d116b56faf7e72c06b8896/src/DCDataSet.cpp#L219) is called.
In a PIConGPU simulation on compute cluster `taurus` at the Dresden University of Technology  were *libSplash* is used for output we run into deadlocks while the dump process.
From different runs with debug output enabled we know that the deadlock is in `DCDataSet::create()`. After disabling the [line](https://github.com/ComputationalRadiationPhysics/libSplash/blob/c536da491a47522502d116b56faf7e72c06b8896/src/DCDataSet.cpp#L219) in splash it looks like everything runs without deadlocks.

The current problem is that we can not reproduce this issue without PIConGPU. With PIConGPU we only know one example with 192 gpus which creates the deadlock.

**Comment by psychocoderHPC** [comment]
My strong suspicion is that [this line](https://github.com/ComputationalRadiationPhysics/libSplash/blob/dev/src/DCDataSet.cpp#L171) is not called with the some values from each mpi rank.
`H5Pset_chunk` must be called with the [same values](https://www.hdfgroup.org/HDF5/doc1.6/CollectiveCalls.html).

Currently I can't see why it should not called with the same values. Maybe some can review `DCDataSet::create()` and the called methods.

**Comment by ax3l** [comment]
@PrometheusPi can you pls add random waits (0.1 - 3s) before calling `writeDomain` in your tests? Maybe we can find a reproducible example here.

E.g., group create (which is collective in parallel) might not be finished for all nodes or similar.

**Comment by ax3l** [comment]
@f-schmitt-zih I am sorry to annoy you with that, do you still remember what's going on in the chunking implementation?

**Comment by ax3l** [comment]
@PrometheusPi can you pls share the test script you build on my PDC example?
Maybe someone has the time to do more tests and we find a reproducible example.

**Comment by ax3l** [comment]
update: errors occured both with checkpoints and outputs.
- `-g 880 4032 4864`
- `-d 2 12 8`
- `-m`

**Comment by psychocoderHPC** [comment]
I remember that I change the ib parameter on clusters with openmpi in the past. The problem was that large mpi calls sometimes creates deadlocks in PIConGPU. Reason for this was that rdma calls greater than the rdma buffer were not split in smaller chunks.  

The current taurus run that crashed not manipulated the open ib buffers maybe that triggered  the deadlock.

**This are the buffer which we that normally for PIConGPU:**
btl_openib_rdma_pipeline_send_length = 100000000
btl_openib_rdma_pipeline_frag_size = 100000000

**Comment by ax3l** [comment]
I can run on tauris with
- 1 node / 4 MPI ranks (gpus: 2 1 2)
- 2 nodes / 8 MPI ranks (gpus: 2 2 2)
- 4 nodes / 16 MPI ranks (gpus: 2 4 2)

but get stderr entries on
- 10 nodes / 40 MPI ranks (ranks: 2 10 2)
```
srun: error: Task launch for 1987247.0 failed on node taurusi2045: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2050: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2048: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2051: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2057: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2049: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2047: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2046: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2052: Header lengths are longer than data received
srun: error: Task launch for 1987247.0 failed on node taurusi2053: Header lengths are longer than data received
srun: error: Application launch failed: Header lengths are longer than data received
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
srun: error: Timed out waiting for job step to complete
```

**Comment by ax3l** [comment]
Since the bug is not occuring any more and likely an IB configuration problem I will close it for now as wontfix.
